eng/scripts/Analyze-AOT-Compact.ps1

```diff
@@ -0,0 +1,189 @@
+    if (Test-Path $projectObjDir) {
+        # Clean project obj directory to ensure fresh build
+        Write-Host "Deleting project obj directory: $projectObjDir"
+        Remove-Item -Path $projectObjDir -Recurse -Force -ErrorAction SilentlyContinue
+    }
+}
+
+function Get-DllNameSet {
+    param(
+        [string]$projectFile
+    )
+    $objDir = Join-Path (Split-Path $projectFile) "bin"
+    $dllNameTable = @{}
+    if (Test-Path $objDir) {
+        $dlls = Get-ChildItem -Path $objDir -Recurse -Filter *.dll -File
+        foreach ($dll in $dlls) {
+            $dllName = $dll.BaseName
+            $dllNameTable[$dllName] = $true
+        }
+    }
+    # Sort: Returns DLL names sorted with the most specialized (most dots) first.
+    return $dllNameTable.Keys |
+        Sort-Object @{ Expression = { $_.Split('.').Count }; Descending = $true },
+                    @{ Expression = { $_ }; Descending = $true }
+}
+
+function Extract-TypeNameFromILWarningLine {
+    param(
+        [string]$line
+    )
+    # Match e.g. 'warning IL2026: Azure.Monitor.OpenTelemetry.AspNetCore.DefaultAzureMonitorOptions.Configure(AzureMonitorOptions): ...'
+    if ($line -match 'warning IL\d+: ([^:]+):') {
+        $full = $Matches[1]
+        # Extract type name before last dot and first parenthesis
+        if ($full -match '^(.*)\.([^.]+)\(') {
+            return $Matches[1]
+        } else {
+            return $full
+        }
+    }
+    return $null
+}
+
+function Find-BestDllMatch {
+    param(
+        [string]$typeName,
+        [string[]]$dllNames # List of DLL names sorted with most specialized first
+    )
+    
+    # Only look for exact matches or prefix matches i.e., typename starts with DLLName
+    # This is the most reliable heuristics for .NET assemblies.
+    foreach ($dllName in $dllNames) {
+        if ($typeName -eq $dllName -or $typeName -like "$dllName.*") {
+            return "$dllName.dll"
+        }
+    }
+    # if the heuristics do not find a match, return "unknown"
+    return "unknown"
+}
+
+function Render-Result {
+    param(
+        [ValidateSet('Html', 'Console')]
+        [string]$OutputFormat = 'Html'
+    )
+    
+    try {
+        & "$PSScriptRoot/Render-AOT-Analysis-Result.ps1" -OutputFormat $OutputFormat
+        if ($OutputFormat -eq 'Html') {
+            Write-Host "HTML report generated: $($config.HtmlReportPath)" -ForegroundColor Green
+        }
+    } catch {
+        Write-Warning "Failed to generate report with format '$OutputFormat': $_"
+    }
+}
+
+## Main program.
+
+Write-Host "Running AOT compatibility (trimming) analysis for runtime: $runtime ..."
+
+# Initialize directories for AOT analysis
+Initialize-AOTAnalysisDirectories -aotReportDir $aotReportDir -projectFile $projectFile
+
+$publishArgs = @(
+    'publish', $projectFile,
+    '--configuration', 'Release',  # Always use Release for AOT analysis
+    '--runtime', $runtime,
+    '--self-contained', 'true',
+    '/p:PublishTrimmed=true',
```

Comment:
This should probably be `PublishAot=true

Reply:
thanks Andy, will check this and will send a follow up pr.